xoai-common/src/main/java/io/gdcc/xoai/model/oaipmh/GetRecord.java

```diff
@@ -31,6 +31,13 @@ public void write(XmlWriter writer) throws XmlWriteException {
         }
     }
 
+    /* Not to be confused with the similarly-named constructor, this
+       method exposes the "private final Record record" to other parts
+       of the OAI framework: */ 
+    public Record getRecord() {
+        return record;
+    }
+    
```

Comment:
The calling graph for this in Dataverse is: `OAIServlet.processRequest()` -> `OAIServlet.writeGetRecord()` -> `XgetRecord.writeToStream()` -> `GetRecord.getRecord()`.

I have no idea why the orignal library intended way wasn't used. (That would be an `XmlWriter` with the `OutputStream` of the servlet, creating the `GetRecord` verb and sending it down the path of the `OAIPMH.write()` path.) This is code back from 2016, so it might remain unknown.

For the way forward: I propose we cleanup the Dataverse side and see what we really need and incorporate those changes in this library. Lots of classes had been forked with minimal changes and it's hard to say why all these things happened as they did...

Reply:
I'll answer/explain this in #2 (or here) tomorrow.